lib/solr.ts

```diff
@@ -583,7 +584,7 @@ update(data: Record<string, any>, options: any, callback?: any) {
    * @api public
    */
 
-search(query, callback) {
+search(query, callback: CallbackFn) {
```

Comment:
type of the first parameter is probably Query

Reply:
Or is it `Query | Record<string, any> | string`? Can you check where we are passing it and what we do with it there?

Reply:
* @param {Query|Object|String} query

I will add type Query | Record<string, any> | string,